internal/app/list/list.go

```diff
@@ -52,14 +56,17 @@ func listConcurrently(item record.RecordEntity, dataFolder string) (chan string,
 	}
 
 	go func() {
-		result, errPrettyPrintDataFile := data.SPrettyPrintFile(
-			dataFolder+string(os.PathSeparator)+item.DataFileName,
+		defer close(c)
```

Comment:
Your code does not read to the end of the channel, but it's a good habit to close channels when you're done writing to them anyway.